### DISCUSSION TITLE
cannot set demand = xx using webgui or HA select field

### DISCUSSION THREAD

**Discussion by turbolooser** [revk/xESP32-Faikout#557] [General]
I know it sounds weird but I cannot set demand = xx using the webgui or the selection menu in Home Assistant.
It gives an error in mqtt and then reverts back to the previous setting on my 3 devices. 

This is the error in mqtt explorer after setting the demand to 100% on the webgui
![image](https://github.com/user-attachments/assets/c909844f-805a-4c67-b11d-cca296c482d2)

Same error if I set this using the selection field in Hassio. Not sure on whats wrong....my automation to set the demand
on all 3 faikins at the same time seems to still work but I am sure I was able to manually override on 1 faikin and the other 2 would accept the new value. I am not really sure why its going like this
![image](https://github.com/user-attachments/assets/7d03aa0f-005d-4e81-b1da-f68d1c05727c)

Anything I could check on my side?
(I also did try to upgrade to a beta but seems the ota is not working right now)

edit1: I have checked my automation which is fine in most cases but even running this command on HA does trigger the mqtt error once in a while....
Thats what I use in my automation to set all 3 faikins to the same demand.... most likely working but still some errors which shouldnot be there I guess?
![image](https://github.com/user-attachments/assets/bd8063b5-95ec-4b7b-a017-bb0c3b7d76c4)

**Comment by revk** [comment]
That usually means the S21 command to the Daikin is not working. You can set `debug` to get some more information.

**Reply by turbolooser** [reply]
I have set the debug on the webui on the Faikin now but not really sure what to look for now when the error is showing up.
Any suggestions?

**Reply by turbolooser** [reply]
I have the feeling that if I do this via my automation on all 3 devices at the same time it works but not if I just change it on 1 device. no clue

**Reply by turbolooser** [reply]
Need to correct my feeling... my automation in sending demand = xx to all 3 faikins also triggers the mqtt error but still the demand control is kind of working. No idea on whats going on and how often this fails but to me this looks like an issue

**Comment by eupesco** [comment]
I sometimes have the same problem, but have not found a resonable solution yet. I suspect a connection to the built-in Daikin WiFi controller - I have seen several times that when the WiFi connection is lost, the demand control values are not taken over from Faikin (but not every time, unfortunately, which makes it hard to debug).
Can you check in Onecta app if the device is offline when the problem occurs? As mentioned above, my not reasonable solution is to switch  the power supply for a complete restart of the devices.

**Reply by turbolooser** [reply]
I have not even connected my devices anymore in Onecta app :-) 
Maybe I need to again

**Reply by turbolooser** [reply]
Connected all 3 devices again but they are all fine on connections. Maybe I should disconnect the daikin wifi module :-)

**Reply by turbolooser** [reply]
I did power cycle also... but no effect.

**Comment by revk** [comment]
I think this will be a question of working out what the Daikin will accept - e.g. sending to just one unit, or to all, and if some specific levels are not acceptable. The Faikin just sends the message - if it does not get a confirmation the value reverts.

**Comment by thomaslunze** [comment]
Could there be any dependecy to this retained flag on status?
Is it supposed to be there? I do see issues now also with using some other settings/messages to the faikin modules reverting back... and I am wondering why. So should the retain flag be set on any topic by default?

![image](https://github.com/user-attachments/assets/e9db5fd4-7830-425d-a40a-e7aaf8e2a686)

**Reply by revk** [reply]
Status is normally retained.

**Comment by thomaslunze** [comment]
Just for my clarity. Any commands/settings being selected in the GUI of Hassio e.g. demand control are being send as topics in the backend on MQTT correct?
Could it make sense to add "QOS=1/2" to these messages to ensure those are beeing received correctly by the faikin?
I am testing this right now for some /settings parameters but for those from the GUI I cannot this right?

**Reply by thomaslunze** [reply]
I am testing again and now realized that this command will do the demand value via mqtt

  topic: faikin wohnzimmer/command/demand
  payload: "95"
which reflects on mqtt explorer like
![image](https://github.com/user-attachments/assets/875f5577-58e3-49a0-8810-38c2b6593fa4)


but as you can see from my error earlier there it says:
![image](https://github.com/user-attachments/assets/1c796eb2-f95f-4de1-84f6-f1f308aa2b6a)

which is not in the right format for a valid faikin command correct?
Any idea on why this messages might have been sent like demand:95 rather than demand=95 ?

**Reply by thomaslunze** [reply]
any idea on this maybe?

**Reply by revk** [reply]
You can send the `command/Faikin/demand` command with a simple numeric value. There is no `=` or `:` involved, the payload is a number.

You can send `command/Faikin/control` with JSON, e.g. `{"demand":95}`

A `failed-set` means the Daikin did not accept the value you are using.

**Reply by dahadd** [reply]
Thanks for the project, but unfortunately

demand control is not set via fainkin, but the values that are set with Onecta-APP are transferred to Faikin

Problems also occur when setting the ECONO mode.
When activated, the switch immediately jumps back to OFF

Unfortunately I have no idea how to create a debug protocol with Faikin

MQTT with HA is set up and works until the problems described above.

**Comment by dahadd** [comment]
Same issue here, demand is not working over the web gui

**Comment by RaW79** [comment]
Have a more general question slightly touching the demand topic. Within the Daikin Onecta App there is a "Auto" setting for the demand.

What happens with these settings in case a Faikin is connected? Is it in general overuled, which means a fix and manual setpoint for demand by the faikin? 

Because the Faikin does not offer a "Auto" setting. Is this Auto mode not available by using a Faikin?
According description of the Onecta App it seems not to be a very satisfying mode but like to understand how the interaction of the these mode settings is foreseen?

Thanks for information!

**Reply by revk** [reply]
Normally the Faikin is passive in that it just accepts any changes it sees from the air-con. It would only override if you change the setting.

**Reply by RaW79** [reply]
Thx. So just in case the Faikin is set to a fix demand value e.g. 75% and then I´ll go to the Onecta App and set it to "Auto" this will activate this mode until I´ll do a change/active command in the Faikin?

Is the Faikin monitoring for this case the damand value? Means when the auto mode is changing the damand by itself the Faikin will auto-update the shown values?

Have not seen such a behavouir yet but this could be because of my ignorance.

**Reply by revk** [reply]
It should

**Reply by revk** [reply]
Unless "auto" is a special setting value in itself which the Faikin does not understand?!

**Reply by RaW79** [reply]
Thanks for clarification. So also the other way around. In case I just want to have full control of the damand by the faikin and no other party doing sth. I should set the demand control in the Onecta App to manual to avoid anything else.

Edit: seems that this does not work. Having the demand control set to manual and then do a demand setting by the faikin and then I see the Daikin App set back to auto-mode automatically.

**Reply by dahadd** [reply]
Reminder:
Thanks for the project, but unfortunately

demand control is not set via fainkin, but the values that are set with Onecta-APP are transferred to Faikin

Problems also occur when setting the ECONO mode.
When activated, the switch immediately jumps back to OFF

Unfortunately I have no idea how to create a debug protocol with Faikin.

MQTT with HA is set up and works until the problems described above.

**Reply by revk** [reply]
OK it it goes off, that means the change sent to air-con has not applied for some reason. It tries any setting a few times but if not reported back as changed it switches back. No idea why it would not apply for you.

**Reply by dahadd** [reply]
and the demand problem? ONCETA to Faikin works, Faikin to ONECTA don't work

**Reply by RaW79** [reply]
> and the demand problem? ONCETA to Faikin works, Faikin to ONECTA don't work

This in general i cannot confirm. Every demand setting done via the faikin (gui or mqtt command) is transfered to the daikin. I can clearly see a reaction to demand changes (power, fan and therfore also compressor RPM).

But I´m not sure how to handle if the auto demand handling by the Onecta App cannot be turned off. I assume there will always be a conflict with my manual demand setting and the auto faikin setting which is unwanted. Only way I see at the moment is continuosly send commands for my manual demand setting and therfore eleminate the daikin setting.

**Reply by dahadd** [reply]
In my system (manual demand control) no value is transferred from Faikin to Daikin, but the other way round it works without any problems. Unfortunately, I have no idea how to analyze the error. is there a protocol at faikin that can be viewed?

**Reply by RaW79** [reply]
Sorry have to correct myself because of this horrorable ONECTA App and very big thanks for the Faikin!

This morning I set the demand by the Onecta App to manual 100% and tried what happens in case of demand changes by the faikin.
The changes have been accepted by the Daikin but the Onecta App jumps back to auto mode. I concluded as result.

So I left it like it is and continued by a manual demand setting by the faikin. Now after hours again I looked to the Onecta App and now the manual 100% is presented/accepted by the App but the hole day was operated in stable manual setting of 70% by the Faikin.

So I´ll continue finding a good approach to control the demand via the Faikin and try not to use the Onecta App as much as possible because this is simply waste of time.

**Reply by MUN0X** [reply]
> Unless "auto" is a special setting value in itself which the Faikin does not understand?!

In the original thread about adding the demand control functionality to faikin, I included how to set the auto mode via the daikin api. I never bothered with looking for the corresponding s21 messages though.

I believe that setting the mode via the daikin api did not trigger a change in D7. At the time I thought that the auto mode is handled by the BRP069-module itself. With the module itself sending D7 commanda at certain times or conditions.  I guess this would also explain the behavior mentioned in this discussion. Perhaps the auto mode is handled elsewhere in s21 but I didnt see it right away. 

Auto mode was shit in my opinion, setting demand control seemingly randomly. 

I pulled the original BRP069C module from my unit and never saw the behavior mentioned in this discussion.

[demand control via daikin api ](https://github.com/revk/ESP32-Faikin/discussions/179#discussioncomment-8100845)
[daikin api to s21](https://github.com/revk/ESP32-Faikin/discussions/179#discussioncomment-8226284)

**Reply by RaW79** [reply]
Thanks and always very intersting to read and understand a bit more.

First/Basically I like to understand how to operate correctly. So its possible to set the demand manually by the faikin gui or mqtt. To make this effective I´ll set the demand control in the Onecta app to On/1  and manual 100%. So in case I like to reduce the demand it will be done by the faikin. The demand value set via the faikin will overule the 100% setting in the app, right?

In case I don´t want demand control, I have to disable it in the Onecta App, right?  And the Faikin (gui or mqtt) will not have any effect, right?

**Reply by MUN0X** [reply]
I removed the original wifi module so I dont know how it interacts/interferes with faikin.

Based on the D7 messages that I captured setting demand to 100 is the same as disabling it.
If I were you, i would just set demand control to manual&100% or off in onecta and then never touch this setting in onecta again.
Just set the demand to 100 with faikin if you want to disable it.

**Reply by RaW79** [reply]
Thanks. Very helpful and will do because of my intention to only operate the demand control via the faikin. The wifi is still in place because I thought it´s nice as additional backup. But it should be clear how to operate and like to avoid unwanted demand control actions.
